lib/phoenix/live_dashboard/live/os_mon_live.ex

```diff
@@ -0,0 +1,264 @@
+defmodule Phoenix.LiveDashboard.OSMonLive do
+  use Phoenix.LiveDashboard.Web, :live_view
+
+  alias Phoenix.LiveDashboard.{
+    SystemInfo,
+    BarComponent,
+    ColorBarComponent,
+    ColorBarLegendComponent
+  }
+
+  @temporary_assigns [system_info: nil, system_usage: nil]
+
+  @cpu_usage_sections [
+    {:kernel, "Kernel", "purple"},
+    {:user, "User", "blue"},
+    {:nice_user, "User nice", "green"},
+    {:soft_irq, "Soft IRQ", "orange"},
+    {:hard_irq, "Hard IRQ", "yellow"},
+    {:steal, "Steal", "purple"},
+    {:wait, "Wait", "orange"},
+    {:idle, "Idle", "dark-gray"}
+  ]
+  @memory_usage_sections [
+    {:in_use_memory, "In use", "purple"},
+    {:buffered_memory, "Buffered", "green"},
+    {:cached_memory, "Cached", "orange"},
+    {:free_memory, "Free", "dark-gray"}
+  ]
+
+  @impl true
+  def mount(%{"node" => _} = params, session, socket) do
+    socket =
+      socket
+      |> assign_defaults(params, session, true)
+      |> assign_system_info()
+
+    {:ok, socket, temporary_assigns: @temporary_assigns}
+  end
+
+  def mount(_params, _session, socket) do
+    {:ok, push_redirect(socket, to: live_dashboard_path(socket, :home, node()))}
+  end
+
+  @impl true
+  def render(assigns) do
+    ~L"""
+    <div class="row">
+      <!-- Left column -->
+      <!-- Cpu information -->
+      <div class="col-sm-6">
+        <h5 class="card-title">
+          CPU
+        </h5>
+        <div class="card mb-4">
+          <div class="card-body resource-usage">
+            <%= for {num_cpu, usage} <- @cpu_per_core do %>
```

Comment:
We should check if `@cpu_per_core == []`, if so, we should skip this particular card. This will be the case on Mac.